Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -658,6 +658,81 @@ class InAppMessageViewTest: IntegrationTest {
         // If url is valid, check if `handleDeepLink` method is called
         XCTAssertFalse(deeplinkUtilMock.handleDeepLinkCalled)
     }
+
+    // MARK: - Send events to Gist event listeners
+
+    // errorWithMessage
+    @MainActor
+    func test_givenAttemptToShowInlineMessage_EventListenerSet_expectFailToShow_expectRouteError() async {
+        messagingInAppImplementation.setEventListener(eventListenerMock)
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessageWithError(givenInlineMessage, insideOfInlineView: inlineView)
+
+        // errorWithMessage called
+        XCTAssertTrue(eventListenerMock.errorWithMessageCalled)
+        XCTAssertEqual(eventListenerMock.errorWithMessageCallsCount, 1)
+        XCTAssertEqual(eventListenerMock.errorWithMessageReceivedArguments, InAppMessage(gistMessage: givenInlineMessage))
+
+        // Other listeners not called
+        XCTAssertFalse(eventListenerMock.messageActionTakenCalled)
+        XCTAssertFalse(eventListenerMock.messageShownCalled)
+        XCTAssertFalse(eventListenerMock.messageDismissedCalled)
+    }
+
+    @MainActor
+    func test_givenAttemptToShowInlineMessageFails_expectMessageNotShown() async {
+        messagingInAppImplementation.setEventListener(eventListenerMock)
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessageWithError(givenInlineMessage, insideOfInlineView: inlineView)
+
+        // Inline message does not display
+        XCTAssertFalse(isInlineViewVisible(inlineView))
+    }
+
+    @MainActor
+    func test_givenAttemptToShowInlineMessage_EventListenerNotSet_expectFailToShow_expectNoCallback() async {
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessageWithError(givenInlineMessage, insideOfInlineView: inlineView)
+
+        // Inline message does not display
+        XCTAssertFalse(isInlineViewVisible(inlineView))
+
+        // errorWithMessage not called as eventListener is not set
+        XCTAssertFalse(eventListenerMock.errorWithMessageCalled)
+    }
+
+    // messageActionTaken when close button is tapped
+    @MainActor
+    func test_givenShowInlineMessage_givenTapCloseButton_expectInlineToHide_expectMessageActionTakenCalled() async {
```

Comment:
```suggestion
    func test_eventListener_givenTapCloseButton_expectCallEventListener() async {
```

The test function and test body makes me think we are testing multiple things in 1 test: 
1. Testing the event listener is called
2. Testing when you click the close button, the View is dismissed. 

Ideally, the test function should only test 1 thing in it. For the scope of this PR, I think you're trying to test the event listener. Therefore, I would edit the test function name to indicate you're only testing the event listener and modify the test body to only test the event listener, not visibility of the View - those tests should be in their own test functions.